runtime/autoload/dist/ft.vim

```diff
@@ -532,6 +532,27 @@ export def FTm()
   endif
 enddef
 
+export def FTmake()
+  # Check if it is a Microsoft Makefile
+  if exists("b:make_microsoft")
+    unlet b:make_microsoft
+  endif
+  var n = 1
+  while n < 1000 && n <= line('$')
+    var line = getline(n)
+    if line =~? '^\s*!\s*\(ifn\=\(def\)\=\|include\|message\|error\)\>'
+      b:make_microsoft = 1
+      break
+    elseif line =~ '^ *ifn\=\(eq\|def\)\>' || line =~ '^ *[-s]\=include\s'
```

Comment:
It's not clear to me yet why you need these two `elseif`s… Other than that, I really have no clue of Microsoft Make, but from what I can tell your patch looks fine.

Reply:
This `elseif` checks GNU Make-style preprocessor directives and the `elseif` in line 548 checks GNU Make-style assignment operators.
These don't happen in Microsoft Make, so we can exit the while loop earlier.